lib/logstash/filters/ruby/script/expect_context.rb

```diff
@@ -0,0 +1,40 @@
+# Handle expect blocks inside of test blocks
+class ExpectContext
+  include ::LogStash::Util::Loggable
+
+  attr_reader :name
+
+  def initialize(test_context, name, block)
+    @test_context = test_context
+    @name = name
+    @block = block
+  end
+
+  def to_s
+    "<Expect #{@test_context.name}/#{self.name}>"
+  end
+
+  def execute(events)
+    begin
```

Comment:
Let's add an assertion here (w/ an actionable error message for the author) that fails if `events` is not an Array containing only  Event objects. The intent is to help folks use the API correctly (since ruby doesn't have types and has implicit returns)

Reply:
I double checked and this is already confirmed in https://github.com/logstash-plugins/logstash-filter-ruby/pull/35/files#diff-ea513e971fcf99f16c937435d6742b63R50.

This `check_result_events!` method checks that the argument is an array of LogStash::Event objects.